src/pycrunch/importing.py

```diff
@@ -53,13 +53,13 @@ def wait_for_batch_status(self, batch, status):
             raise ValueError("The batch did not reach the '%s' state in the "
                              "given time. Please check again later." % status)
 
-    def add_source(self, ds, filename, fp, mimetype):
+    def add_source(self, ds, filename, fp, mimetype, data={}):
```

Comment:
Best to default to None - `data=None`, in this context an empty dict is probably safe, but it is considered to be a bad smell.

Reply:
Good catch, definitely a bug waiting to happen